addons/website_slides/models/slide_slide.py

```diff
@@ -266,6 +274,19 @@ def _compute_category_id(self):
                 elif slide.category_id != current_category:
                     slide.category_id = current_category.id
 
+    @api.depends('slide_type', 'question_ids', 'channel_id.is_member')
+    @api.depends_context('uid')
+    def _compute_member_actions(self):
+        """Determine if the slide can be marked as (un)completed."""
+        for slide in self:
+            slide.can_be_undone = slide.website_published and slide.channel_id.is_member
```

Comment:
This one should probably also check the quiz stuff.

Also, the mark_completed / mark_uncompleted should respectively check that the slide is not already marked complete / incomplete.